lib/simple_token_authentication.ex

```diff
@@ -12,19 +12,19 @@ defmodule SimpleTokenAuthentication do
 
   def call(conn, _opts) do
     val = get_auth_header(conn)
-    
-    tokens =
-      :simple_token_authentication
-      |> Application.get_env(:token)
-      |> List.wrap()
-      
-    service_tokens =
-      :simple_token_authentication
-      |> Application.get_env(:service_tokens)
-      |> List.wrap()
 
-    if Enum.any?(tokens ++ service_tokens, &matches?(&1, val)) do
-      conn
+    token_map =
+      case :persistent_term.get(:simple_token_authentication_map, nil) do
+        nil ->
+          build_token_map()
```

Comment:
This doesn't seem like that intensive an operation, but would it make more sense to do it at app startup than on the first call?

Reply:
I was debating about that. maybe something we can do later? not sure it will make a big difference since we still need to ensure that it exist and do something about it?